pkg/ami/auto_resolver.go

```diff
@@ -48,10 +49,13 @@ var ImageSearchPatterns = map[string]map[string]map[int]string{
 	},
 }
 
-// ImageFamilyToAccountID is a map of image families to account Ids
-var ImageFamilyToAccountID = map[string]string{
-	ImageFamilyAmazonLinux2: "602401143452",
-	ImageFamilyUbuntu1804:   "099720109477",
+// accountIDResolverFunc resolves ami provider accountID by region
+type accountIDResolverFunc func(region string) string
+
+// ImageFamilyToAccountID is a map of image families to accountIDResolverFunc
+var ImageFamilyToAccountID = map[string]accountIDResolverFunc{
```

Comment:
Perhaps now that we need a function anyway it makes more sense to get rid of the map and make it a function:

```
accountId(imageFamily, region) (string, error) {
  switch imageFamily {
    case ImageFamilyUbuntu1804:
      return "099720109477", nil
    case ImageFamilyAmazonLinux2:
      return api.EKSResourceAccountID(region), nil
    default:
      return nil, new error....
  }
}
```

Reply:
Then we can unit test it

Reply:
make sense :D, modified